### PR TITLE
`tdata` to `teal_data`

### PR DESCRIPTION
**Pull request by gogonzo**
This is a feature branch, **DON'T MERGE and DON'T COMMENT**. PR open to have a constant checks. We will start review this PR as a whole in the sprint 4.

Summary
- removed all occurrences of `TealData` and friends (enemies),
- removed dependency on `teal.slice`
- fixed `data_extract_multiple_srv` and `merge_expression_srv` to work with (reactive) `teal_data` https://github.com/insightsengineering/teal.transform/issues/159,
- fixed `data_extract_ui` to be independent from data.  Now, `data_extract_ui` can be initialized with "delayed" specs and choices/selected are updated by the server on init. https://github.com/insightsengineering/teal.transform/issues/111 
- removed deprecated data_merge_module https://github.com/insightsengineering/teal.transform/issues/144

### CODE REVIEW

**Review by chlebowa — CHANGES_REQUESTED** [review]
blocked

File: R/data_extract_module.R [hunk 1]
```diff
@@ -655,15 +647,23 @@ data_extract_srv.list <- function(id, datasets, data_extract_spec, join_keys = N
...
     }
   })
-  checkmate::assert_multi_class(datasets, classes = c("FilteredData", "list"))
   UseMethod("data_extract_multiple_srv", datasets)
 }
 
+#' @rdname data_extract_multiple_srv
+#' @export
+data_extract_multiple_srv.reactive <- function(data_extract, datasets, ...) {
+  # convert teal_data to list of reactives
```
Comment (chlebowa) [inline R/data_extract_module.R:662]:
```suggestion
  # convert reactive containing teal_data to list of reactives with one dataset each
```

File: R/resolve.R [hunk 2]
```diff
@@ -175,7 +175,7 @@ resolve.default <- function(x, datasets, keys) {
 #'
 #' @param x (`function`) Function that is applied on dataset.
 #' It must take only a single argument "data" and return character vector with columns / values.
-#' @param ds (`data.frame`) `TealDataset` on which the function is applied to.
+#' @param ds (`data.frame`) dataset on which the function is applied to.
```
Comment (chlebowa) [inline R/resolve.R:178]:
```suggestion
#' @param ds (`data.frame`)
```

File: tests/testthat/test-data_extract_multiple_srv.R [hunk 3]
```diff
@@ -45,60 +38,63 @@ testthat::test_that("data_extract_multiple_srv returns a named reactive list wit
...
+    class = "FilteredData"
   )
-
   shiny::withReactiveDomain(
     domain = shiny::MockShinySession$new(),
     expr = testthat::expect_error(
-      data_extract_multiple_srv(data_extract = list(test = NULL), datasets = nr_data_list, join_keys = key_list),
+      data_extract_multiple_srv(data_extract = list(test = NULL), datasets = mock_datasets),
       regexp = NA
     )
```
Comment (chlebowa) [inline tests/testthat/test-data_extract_multiple_srv.R:83]:
```suggestion
    expr = testthat::expect_no_error(
      data_extract_multiple_srv(data_extract = list(test = NULL), datasets = mock_datasets)
    )
```

File: tests/testthat/test-data_extract_srv.R [hunk 4]
```diff
@@ -44,58 +28,65 @@ adlb_extract <- data_extract_spec(
...
+    domain = shiny::MockShinySession$new(),
+    expr = testthat::expect_error(
+      data_extract_srv(
+        id = "x",
+        data_extract_spec = adsl_extract,
+        datasets = data_list,
+        join_keys = join_keys
+      ),
+      NA
     )
```
Comment (chlebowa) [inline tests/testthat/test-data_extract_srv.R:71]:
```suggestion
    expr = testthat::expect_no_error(
      data_extract_srv(
        id = "x",
        data_extract_spec = adsl_extract,
        datasets = data_list,
        join_keys = join_keys
      )
   )
```

File: R/resolve.R [hunk 5]
```diff
@@ -175,7 +175,7 @@ resolve.default <- function(x, datasets, keys) {
 #'
 #' @param x (`function`) Function that is applied on dataset.
 #' It must take only a single argument "data" and return character vector with columns / values.
-#' @param ds (`data.frame`) `TealDataset` on which the function is applied to.
+#' @param ds (`data.frame`)
```
[the remark this answers is not in the document]
Comment (averissimo) [reply R/resolve.R:178]: hahahaha true, but it should say something, having just the datatype seems odd.

Even if it just unpacks the abbreviation :shrug: 

```suggestion
#' @param ds (`data.frame`) Dataset.
```

**Review by chlebowa — APPROVED** [review]

**Pull request merged**